hadoop-tools/hadoop-azure/src/main/java/org/apache/hadoop/fs/azurebfs/services/AbfsInputStream.java

```diff
@@ -582,6 +701,29 @@ int readRemote(long position, byte[] b, int offset, int length, TracingContext t
     return (int) bytesRead;
   }
 
+  private void initPathPropertiesFromReadPathResponseHeader(final AbfsHttpOperation op) throws IOException {
+    if (DIRECTORY.equals(
+        op.getResponseHeader(HttpHeaderConfigurations.X_MS_RESOURCE_TYPE))) {
+      throw new FileNotFoundException(
+          "read must be used with files and not directories. Path: " + path);
+    }
+    contentLength = parseFromRange(
+        op.getResponseHeader(HttpHeaderConfigurations.CONTENT_RANGE));
+    eTag = op.getResponseHeader(HttpHeaderConfigurations.ETAG);
+    fileStatusInformationPresent.set(true);
```

Comment:
Should we have a check here that contentLength and ETag are valid before setting it to true??

Or is it okay to have it -1??
I see some handling done for that already.
Just a thought

Reply:
Content_range is expected to be in format <response_range_start>-<response_range_end>/<contentLength>.
In case if server doesnt follow this format, content-range would be marked as -1. Keeping it as -1 would stop future reads on the inputStream from happening.

Reply:
But, there could be things like we got 409 (CPK failure), in that case content_range does not come. I think we can have a check if the response headers contain both contentRange and eTag, we should be good, else, we will not set.

I am in sync with you , will take it!